packages/block-editor/src/store/test/private-selectors.js

```diff
@@ -21,4 +25,186 @@ describe( 'private selectors', () => {
+					[
+						'b26fc763-417d-4f01-b81c-2ec61e14a972',
+						'ef45d5fd-5234-4fd5-ac4f-c3736c7f9337',
+					],
+					[
+						'9b9c5c3f-2e46-4f02-9e14-9fe9515b958f',
+						'ef45d5fd-5234-4fd5-ac4f-c3736c7f9337',
+					],
+					[
+						'b3247f75-fd94-4fef-97f9-5bfd162cc416',
+						'9b9c5c3f-2e46-4f02-9e14-9fe9515b958f',
+					],
+					[
+						'e178812d-ce5e-48c7-a945-8ae4ffcbbb7c',
+						'9b9c5c3f-2e46-4f02-9e14-9fe9515b958f',
+					],
+				] ),
+			},
+			blockListSettings: {
+				'ef45d5fd-5234-4fd5-ac4f-c3736c7f9337': {},
+				'9b9c5c3f-2e46-4f02-9e14-9fe9515b958f': {},
+			},
+			blockEditingModes: new Map( [] ),
+		};
+
+		const __experimentalHasContentRoleAttribute = jest.fn( () => false );
+		getBlockEditingMode.registry = {
+			select: jest.fn( () => ( {
+				__experimentalHasContentRoleAttribute,
+			} ) ),
+		};
+
+		it( 'should return default by default', () => {
+			expect(
+				getBlockEditingMode(
+					baseState,
+					'b3247f75-fd94-4fef-97f9-5bfd162cc416'
+				)
+			).toBe( 'default' );
+		} );
+
+		[ 'disabled', 'contentOnly' ].forEach( ( mode ) => {
+			it( `should return ${ mode } if explicitly set`, () => {
+				const state = {
+					...baseState,
+					blockEditingModes: new Map( [
+						[ 'b3247f75-fd94-4fef-97f9-5bfd162cc416', mode ],
+					] ),
+				};
+				expect(
+					getBlockEditingMode(
+						state,
+						'b3247f75-fd94-4fef-97f9-5bfd162cc416'
+					)
+				).toBe( mode );
+			} );
+
+			it( `should return ${ mode } if explicitly set on a parent`, () => {
+				const state = {
+					...baseState,
+					blockEditingModes: new Map( [
+						[ 'ef45d5fd-5234-4fd5-ac4f-c3736c7f9337', mode ],
+					] ),
+				};
+				expect(
+					getBlockEditingMode(
+						state,
+						'b3247f75-fd94-4fef-97f9-5bfd162cc416'
+					)
+				).toBe( mode );
+			} );
+
+			it( `should return ${ mode } if overriden by a parent`, () => {
+				const state = {
+					...baseState,
+					blockEditingModes: new Map( [
+						[ '', mode ],
+						[ 'ef45d5fd-5234-4fd5-ac4f-c3736c7f9337', 'default' ],
+						[ '9b9c5c3f-2e46-4f02-9e14-9fe9515b958f', mode ],
+					] ),
+				};
+				expect(
+					getBlockEditingMode(
+						state,
+						'b3247f75-fd94-4fef-97f9-5bfd162cc416'
+					)
+				).toBe( mode );
+			} );
+
+			it( `should return ${ mode } if explicitly set on root`, () => {
+				const state = {
+					...baseState,
+					blockEditingModes: new Map( [ [ '', mode ] ] ),
+				};
+				expect(
+					getBlockEditingMode(
+						state,
+						'b3247f75-fd94-4fef-97f9-5bfd162cc416'
+					)
+				).toBe( mode );
+			} );
+		} );
+
+		it( 'should return disabled if parent is locked and the block has no content role', () => {
+			const state = {
+				...baseState,
+				blockListSettings: {
+					...baseState.blockListSettings,
+					'9b9c5c3f-2e46-4f02-9e14-9fe9515b958f': {
+						templateLock: 'contentOnly',
+					},
+				},
+			};
+			__experimentalHasContentRoleAttribute.mockReturnValueOnce( false );
```

Comment:
I suppose this isn't necessary because `blockTypeName` is always going to be `undefined` (and `__experimentalHasContentRoleAttribute`  therefore will return `false`)

Reply:
True but I like the explicitness.

Reply:
You do you. 👍